File: WindowsMobileClient/Windows Mobile Client/Classes/Clipboard.vb

Public NotInheritable Class Clipboard
    <System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1024:UsePropertiesWhereAppropriate")> _
    Public Shared Function GetText() As String
        Dim o As IDataObject = Windows.Forms.Clipboard.GetDataObject()
        If o IsNot Nothing AndAlso o.GetDataPresent(DataFormats.StringFormat) Then
            Return CStr(o.GetData(DataFormats.StringFormat, True))
        Else
            Return String.Empty
        End If
    End Function

    Public Shared Sub SetText(ByVal value As String)
        Windows.Forms.Clipboard.SetDataObject(value)
    End Sub

    Private Sub New()

    End Sub
End Class

File: Library/activiser.Library.Forms/Clipboard.cs
using MSForms = System.Windows.Forms;

namespace activiser.Library.Forms
{
    public sealed class Clipboard
    {
        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1024:UsePropertiesWhereAppropriate")]
        public static string GetText()
        {
            MSForms.IDataObject o = MSForms.Clipboard.GetDataObject();
            if (o != null && o.GetDataPresent(MSForms.DataFormats.StringFormat))
            {
                return (string)o.GetData(MSForms.DataFormats.StringFormat, true);
            }
            else
            {
                return string.Empty;
            }
        }

        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1024:UsePropertiesWhereAppropriate")]
        public static void SetText(string value)
        {
            MSForms.Clipboard.SetDataObject(value);
        }

        private Clipboard()
        {

        }
    }
}
